dask/dataframe/tests/test_dataframe.py

```diff
@@ -1882,6 +1882,13 @@ def test_repartition_on_pandas_dataframe():
     assert_eq(ddf, df.y)
 
 
+def test_repartition_gh6543():
+    df = pd.DataFrame({"a": np.arange(16)})
+    a = dd.from_pandas(df, chunksize=4)
+    b = a.repartition(npartitions=16)
+    assert b.npartitions == 15
```

Comment:
I don't think that we want to test that there are 15 partitions.  Ideally there would be 16, and they would all be non-empty.  If someone were to implement a fix for this then we wouldn't want this test to tell them that they were wrong.  

Instead, I think that your goal is to ensure that none of the partitions are empty.  Maybe something like this?

```python
lengths = b.map_partitions(len).compute()
assert all(list(lengths))
```

Reply:
That should handle the check that none are empty. Succinct!